### PR TITLE
Improve the error messages for `checks.go`

### PR DESCRIPTION
**Pull request by iwahbe**
I discovered this when working on
https://github.com/pulumi/pulumi-terraform-bridge/pull/2245. This improves the quality of the error message for some cases, making it more actionable.

This PR also allows users to override the `Secret` value to remove the sensitive error. This is logically consistent with `.Type` overrides.

### CODE REVIEW

File: pf/internal/check/check_test.go [hunk 1]
```diff
@@ -90,14 +127,26 @@ func test(t *testing.T, info tfbridge.ProviderInfo) (string, error) {
 }
 
 type (
-	testProvider struct{ provider.Provider }
-	testResource struct{ resource.Resource }
+	testProvider struct {
+		provider.Provider
+
+		missingID, sensativeID bool
```
Comment (guineveresaenger) [inline pf/internal/check/check_test.go:133]:
```suggestion
		missingID, sensitiveID bool
```

File: pf/internal/check/checks.go [hunk 2]
```diff
@@ -78,26 +75,79 @@ func checkIDProperties(sink diag.Sink, info tfbridge.ProviderInfo, isPFResource
...
+	}
+	return msg
+}
+
+type errSensativeID struct {
+	token string
+}
+
+func (err errSensativeID) Error() string {
+	msg := `"id" attribute is sensitive, but cannot be kept secret.`
```
Comment (guineveresaenger) [inline pf/internal/check/checks.go:96]: Would it be worth it to say why it cannot be kept secret?

**Review by guineveresaenger — APPROVED** [review]
I don't generally block on spelling, but I'd really love it if all instances of "sensitive" were spelled the same. 🙏

File: pf/internal/check/check_test.go (hunk 1 above)
Comment (iwahbe) [reply pf/internal/check/check_test.go:133]: > I don't generally block on spelling

Please do let me know if you spot a spelling mistake. I'll fix it before merging!

File: pf/internal/check/checks.go (hunk 2 above)
Comment (iwahbe) [reply pf/internal/check/checks.go:96]: The best explanation for why is https://github.com/pulumi/pulumi/pull/8948, but it's not a very good explanation. I'll open an issue.

Comment (iwahbe) [reply pf/internal/check/checks.go:96]: https://github.com/pulumi/pulumi/issues/16818 is a slightly better explanation.

**Pull request merged**